### ISSUE TITLE
Temporary units inside an equation

### ISSUE THREAD

**Issue by Mobz87** [angularsen/UnitsNet#514]
If I set up the following equation:
Entropy = Entropy + SpecificEnergy* ( Entropy / SpecificEnergy )

I get the error:
Operator '/' cannot be applied to operands of type Entropy and SpecificEnergy

I guess the reason is that "" ( Entropy / SpecificEnergy ) "" is not equal to any known unit. But math-wise it is perfectly fine.
Is this a general thing or am I just using it wrong? it would be nice to be able to do any kind of equation

**Comment by angularsen** [comment]
Hi @Mobz87 , would it help to rephrase this to
`Entropy = Entropy + (SpecificEnergy1 / SpecificEnergy2) * Entropy` ?

Then it could work if we added a division overload like this: 
`double ratio = SpecificEnergy / SpecificEnergy`

**Comment by Mobz87** [comment]
Hi @angularsen 

Thanks for the quick answer!
I guess that could work, I will for sure try it

When you have a large amount of eqs you dont want to redo them all too much.

Would it be impossible for it to do some logic on the units it self?

I mean in this case I try to: 
` ( Entropy / SpecificEnergy )`
` ( J / K )  /  ( J / Kg)`
which gives
`(J * Kg) / (K * Kg)`
With a little bit of logic:
`J/K`

This might not be a registered units (which is why it fails i guess) but if it tried to calculate the whole equation with this Unit logic it would see that everything is fine.

**Comment by angularsen** [comment]
The problem is that we rely on C# language's operator overloads for this, and it is simply not intended to optimize/reduce complex expressions like this. It handles pairs of `+` and `*` and `/` expressions at a time, so the only way I can see this working is to ensure there exist operator overloads for all quantity combinations you want to use. Even when it produces _temporary_ quantities that don't make sense or have a name.

I can imagine this is possible, but I don't immediately see the solution. Maybe if we had a way to represent quantities generically, defined only by its dimensions in the 7 SI base units (we already have the `BaseDimensions` property), then we could find some clever way to achieve this.

**Comment by tmilnthorp** [comment]
I plan do to just that at some point. Haven't had a chance to look recently!

**Comment by Mobz87** [comment]
I agree that having types defining every combinations is a losing battle.
@tmilnthorp That sounds really great! Looking forward to that!

**Comment by Mobz87** [comment]
> Hi @Mobz87 , would it help to rephrase this to
> `Entropy = Entropy + (SpecificEnergy1 / SpecificEnergy2) * Entropy` ?
> 
> Then it could work if we have operator overloads like this (not sure if we have them or not):
> `double ratio = SpecificEnergy / SpecificEnergy`
> and
> `Entropy = double * Entropy`

It did work!

**Comment by angularsen** [comment]
Awesome 👏

**Comment by angularsen** [comment]
I'll close this issue, @tmilnthorp will start a new conversation about the dynamic operator overload proposal.
